sentry-android-core/src/main/java/io/sentry/android/core/AndroidUserCache.java

```diff
@@ -0,0 +1,90 @@
+package io.sentry.android.core;
+
+import android.content.SharedPreferences;
+import io.sentry.core.IUserCache;
+import io.sentry.core.protocol.User;
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// TODO: to think, this could be a SentryOptionsCache that could be expanded, does it make sense?
```

Comment:
to discuss

Reply:
Makes sense to me to keep this focused on user caching, specially the abstraction

Reply:
I'm not sure if Cache is the right naming here, because we actually persist the data to the `SharedPreferences`. A cache is usually used to improve the performance of something, see CPU caches, browser caches, web server caches. For me it is more like a `UserRepo` or `UserStorage`. What do you think?

Reply:
not entirely, I borrowed the idea from https://developer.android.com/reference/android/content/Context.html#getCacheDir()

I see the Caching term as something temporary, being in memory or in the disk.
Also, https://docs.sentry.io/clients/java/integrations/#android (old SDK version)

> Events will be buffered to disk (in the application’s cache directory) by default

I'm fine changing it as well, but it might make it inconsistent with other SDKs.

@bruno-garcia opnions?